src/test/java/org/jabref/logic/bibtex/comparator/PreambleDiffTest.java

```diff
@@ -0,0 +1,46 @@
+package org.jabref.logic.bibtex.comparator;
+
+import java.util.Optional;
+
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseContext;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PreambleDiffTest {
+
+    private final BibDatabaseContext originalDataBaseContext = mock(BibDatabaseContext.class);
+    private final BibDatabaseContext newDataBaseContext = mock(BibDatabaseContext.class);
+    private final BibDatabase originalDataBase = mock(BibDatabase.class);
+    private final BibDatabase newDataBase = mock(BibDatabase.class);
+
+    @BeforeEach
+    void setUp() {
+        when(originalDataBaseContext.getDatabase()).thenReturn(originalDataBase);
+        when(newDataBaseContext.getDatabase()).thenReturn(newDataBase);
+    }
+
+    @Test
+    void compareSamePreambleTest() {
+        when(originalDataBase.getPreamble()).thenReturn(Optional.of("preamble"));
+        when(newDataBase.getPreamble()).thenReturn(Optional.of("preamble"));
+
+        assertEquals(Optional.empty(), PreambleDiff.compare(originalDataBaseContext, newDataBaseContext));
+    }
+
+    @Test
+    void compareDifferentPreambleTest() {
+        when(originalDataBase.getPreamble()).thenReturn(Optional.of("preamble"));
+        when(newDataBase.getPreamble()).thenReturn(Optional.of("otherPreamble"));
+
+        PreambleDiff expected = new PreambleDiff(Optional.of("preamble").orElse(""), Optional.of("otherPreamble").orElse(""));
+        PreambleDiff result = PreambleDiff.compare(originalDataBaseContext, newDataBaseContext).orElse(null);
+        assertEquals(expected.getOriginalPreamble(), result.getOriginalPreamble());
+        assertEquals(expected.getNewPreamble(), result.getNewPreamble());
```

Comment:
Why not `assertEquals(expected, result)`. Does something at the equalsmethod not work? - Maybe, you have to implement `equals` to get the tests running?